### DISCUSSION TITLE
Yamaha RX-V573 Compatibility and Zone Control

### DISCUSSION THREAD

**Discussion by McClutcheon** [mvdwetering/yamaha_ynca#321] [General]
I have been able to successfully configure my RX-V573 with no issues, I noticed it wasn't on the list of compatible devices so figured I would share. 

Noteworthy items for this model is that it does offer "two zones" however not in the typical manner. The secondary zone is simply a second set of speakers that share the same input and volume as the main zone. They are enables by a Zone B button on the remote OR you can use scenes to define the input and the "zones". This is described in greater detail at this post:  https://community.home-assistant.io/t/yamaha-rx-573-receiver-with-zone-b-speakers/36948

Basically they use scenes to act as a zone control, and use a python script to query the receiver for the input prior to scene selection, since when the scene is selected the input will change. This allows the input to be changed back after the scene has been used to enable/disable the zone. I was also able to get this working, with an input select, and some automations in home assistant. Not sure if this would be useful / could get integrated but I can confirm it works 100%.

**Comment by mvdwetering** [comment]
I added the RX-V573 to the readme, thanks 👍 

I don't recall seeing commands to controlling such A/B zones, but we can try to figure it out.

Could you enable/disable the zones (both off, A on, both on, B on) and then get the diagnostics from the integration. After that attach the file (don't paste the contents to keep the comments readable).
It will contain the most recent commands received from the receiver. Maybe we can figure out the commands to control the A/B zones from that.

You can download the diagnostics from the integration entry in the menu. Sorry about the Dutch menus, I hope it is clear enough. In English is says "Download diagnostics" I believe.
![image](https://github.com/user-attachments/assets/4e515ddc-5673-40c8-a3fe-0c8f8014213e)

**Comment by McClutcheon** [comment]
Not sure if this is applicable or not but I dont see any zone controls for this receiver (or my 583)  when checking the available commands:(http://myIP_for_receiver/YamahaRemoteControl/desc.xml). As I mentioned above I have been able to achieve the zone control functionality through means of the scene selection. I have it setup so that Scene 1 is only Zone A, Scene 2 only Zone B, Scene 3 Zone A & B. Im not sure if there is any options to select the scene through this integration at this time.

I do see scene selection commands for both my receivers so this does seem possible to integrate, and with a little more work we could likely integrate retaining the source input across scenes using the logic in the original post. Im still waiting on the trigger source prior to collecting the diagnostics as I mentioned in my Issue post. Once I get that info I will come back with the diagnostic data.

**Comment by mvdwetering** [comment]
This integration uses the Yamaha YNCA protocol which is different from MusicCast or what the default Yamaha integration in Home Assistant is using. It is a completely separate protocol and It does not use information from the `desc.xml` file.

---

(edit: reading your comment again I think I misunderstood and you already have the scene buttons, but leaving it here just in case)

You might be able to select scenes with the current integration. 

For most receivers the integration will auto detect if it supports the Scene commands in the protocol. It wil then add scene buttons like in the screenshot below (names could be different). If these buttons are not there your receiver probably does not support the scene commands. 
But you can try to ignore the autodetection and force the amount of scenes in the integration configuration by pressing "CONFIGURE" on the integration entry and then go through the flow. Forcing scene buttons this way does not mean they will work, it just means they will  always be created.  But it is an easy way to check and if it does not work just set it back to Auto Detect or 0 .

![image](https://github.com/user-attachments/assets/9545daed-2043-4f97-8826-a2a1496db6a7)

![image](https://github.com/user-attachments/assets/4d3bd39c-9f5a-47a9-8866-6b56377894bd)

---

Another way to trigger scenes is to send the SCENE# commands through the remote entity. You can try it in the Developer Tools by selecting the "Remote: Send Command" service/action, then select the remote entity and fill in the command you want to send e,g, "scene_1". If it works you can send the command from automations or you could create Template Buttons from the Helpers section in the Home Assistant settings.

The `yaml` is shown below.

```
action: remote.send_command
data:
  command: scene_1
target:
  entity_id: remote.rx_a810_main_remote
```

In the UI it looks like this:
![image](https://github.com/user-attachments/assets/05c44119-bf53-4801-8418-c5fb80933699)

**Comment by McClutcheon** [comment]
[config_entry-yamaha_ynca-01J5KA0CK5YCY41D8VB8Z5DTS2.json](https://github.com/user-attachments/files/16650910/config_entry-yamaha_ynca-01J5KA0CK5YCY41D8VB8Z5DTS2.json)
Attached diagnostic data from RX-V573 after using IR remote to trigger Zone A/B on/off. 

I took a quick peek and the scene selections were autodetected, my apologies for being oblivious. They work properly which is no issue, I just heavily rely on my current script which uses POST commands with the xml data for retaining the input after the scene selection. Any ideas on using the scene selection to trigger an automation for setting the source input back after a scene selection?

**Reply by mvdwetering** [reply]
Sorry for late reply, I was initially focused on the RX-V583.

It seems like the diagnostics from RX-V583 got attached by accident. 
Would it be possible to attach the one from RX-V573?

**Comment by McClutcheon** [comment]
[config_entry-yamaha_ynca-01J5K3BZE7GK3AFG1QS173KES0.json](https://github.com/user-attachments/files/16741174/config_entry-yamaha_ynca-01J5K3BZE7GK3AFG1QS173KES0.json)
My apologies, this should be the correct one

**Reply by mvdwetering** [reply]
I am not seeing anything related to ZoneB in the diagnostics.

In case you downloaded a new diagnostics file from Home Assistant then probably the Zone toggle commands you did are too long ago.

Anyway, I kind of suspect the Zone A/B toggles will work with the prerelease/beta I just put out as mentioned in the other topic. Lets see what the results are before getting new diagnostics.
